sqllogictest/src/parser.rs

```diff
@@ -638,11 +758,32 @@ fn parse_inner<T: ColumnType>(loc: &Location, script: &str) -> Result<Vec<Record
                 // Lines following the "----" are expected results of the query, one value per line.
                 let mut expected_results = vec![];
                 if has_result {
-                    for (_, line) in &mut lines {
-                        if line.is_empty() {
-                            break;
+                    if let Some(e) = &expected_error {
+                        if e.is_empty_inline() {
+                            let mut results = String::new();
+                            while let Some((_, line)) = lines.next() {
+                                // 2 consecutive empty lines
+                                if line.is_empty()
+                                    && lines.peek().map(|(_, l)| l.is_empty()).unwrap_or(true)
+                                {
+                                    lines.next();
+                                    break;
+                                }
+                                results += line;
+                                results.push('\n');
+                            }
+                            expected_error =
+                                Some(ExpectedError::Multiline(results.trim().to_string()));
+                        } else {
+                            return Err(ParseErrorKind::DuplicatedErrorMessage.at(loc.clone()));
+                        }
```

Comment:
We may put these in a function (`parse_multiline_error`?). It's identical with above.